[quantization] state_dict serialization for Conv2d + some bugfixes

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24117 [quantization] JIT serialization for Conv2d
* **#24116 [quantization] state_dict serialization for Conv2d + some bugfixes**
* #24115 [quantization] Re-work Conv2d

Differential Revision: [D16765476](https://our.internmc.facebook.com/intern/diff/D16765476)

Commit:
[quantization] state_dict serialization for Conv2d + some bugfixes

raghuramank100 commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -103,6 +103,42 @@ def forward(self, input):
+            self.bias.copy_(state_dict[prefix + 'bias'])
+            state_dict.pop(prefix + 'bias')
+        else:
+            self.bias = None
```
Do we need special handling of self.bias in get_state and set_state functions?

jamesr66a replied:
Actually, I'm going to remove the special handing in these functions since I just discovered it's not needed

raghuramank100 reviewed (approved):
Looks good, had a comment on handling bias for serialization after scripting.

Commit:
Update on "[quantization] state_dict serialization for Conv2d + some bugfixes"

Commit:
Update on "[quantization] state_dict serialization for Conv2d + some bugfixes"


Differential Revision: [D16765476](https://our.internmc.facebook.com/intern/diff/D16765476)

Commit:
Update on "[quantization] state_dict serialization for Conv2d + some bugfixes"


Differential Revision: [D16765476](https://our.internmc.facebook.com/intern/diff/D16765476)
